### ISSUE TITLE
Config to allow `undefined` in typescript server output

### ISSUE THREAD

**Issue by alsiola** [dotansimha/graphql-code-generator#1012]
**Is your feature request related to a problem? Please describe.**

Generated types are not-compatible with other libraries typings that use undefined instead of null.  For example, the generated type for a Query response might be:

```
export interface FeatureResponse {
  feature?: Feature | null;
}
```
If am using TypeORM (for example) and resolve this using the generated resolver:

```
const feature: QueryResolvers.FeatureResolver = (_, { id }) => {
    return repository.findOne({ id });
};
```
Then this will not compile. This is because the `entity.findOne` function returns `Promise<Feature | undefined>` whereas the generate resolver requires `Feature | null | Promise<Feature | null>`.
In order to use the generated types it is necessary to convert undefined values to null in the resolver, which is repetitive and noisy.

**Describe the solution you'd like**
It would be useful to allow configuration to specify how optional values should be represented. For example, this might look like:

```
schema: http://localhost:4000/graphql
overwrite: true
generates:
  src/types.ts:
    - typescript-common
    - typescript-server
    - typescript-resolvers
config:
  contextType: ./#Ctx
  optionalType: undefined
```

**Describe alternatives you've considered**
The other alternative that springs to mind is a plugin, but this would seem to duplicate the work there, and would require continued effort to maintain a separate plugin that is 99.9% identical to the original.

Many thanks for this excellent project!

**Comment by kamilkisiela** [comment]
@dotansimha #1017 and #1018 should enable that

**Comment by dotansimha** [comment]
Available since 0.15.0 :)

**Comment by alsiola** [comment]
That's awesome, thanks for your continued work on this great project!
